npm install fails: GitHub redirects release downloads to release-assets.githubusercontent.com

## Summary

Fresh `npm install -g git-orchard` fails on Linux (and likely macOS) with:

\`\`\`
npm ERR! Failed to download orchard binary: Redirect to untrusted host: release-assets.githubusercontent.com
\`\`\`

## Root cause

`npm/install.js` maintains an allowlist of hosts for redirect targets (defense against malicious redirects):

\`\`\`js
const ALLOWED_HOSTS = new Set([
  'github.com',
  'objects.githubusercontent.com',
  'github-releases.githubusercontent.com',
]);
\`\`\`

GitHub now redirects release asset downloads to \`release-assets.githubusercontent.com\`, which is not on the list. The redirect is rejected at \`install.js:82\`.

## Reproduction

On a fresh Ubuntu 24.04 VM (orchard.boxd.sh), with node 18 + npm 9:

\`\`\`
sudo npm install -g git-orchard
# npm ERR! Failed to download orchard binary: Redirect to untrusted host: release-assets.githubusercontent.com
\`\`\`

## Fix

Add \`release-assets.githubusercontent.com\` to \`ALLOWED_HOSTS\` in \`npm/install.js\`.

## Acceptance Criteria

- [ ] \`ALLOWED_HOSTS\` includes \`release-assets.githubusercontent.com\`
- [ ] \`npm install -g git-orchard\` succeeds on a fresh Ubuntu 24.04 VM
- [ ] Checksum verification still works end-to-end

--- a/npm/install.js
+++ b/npm/install.js
@@ -40,6 +40,7 @@ const ALLOWED_HOSTS = new Set([
   'github.com',
   'objects.githubusercontent.com',
   'github-releases.githubusercontent.com',
+  'release-assets.githubusercontent.com',
 ]);
 
 const MAX_REDIRECTS = 5;

